src/Oro/Bundle/EntityBundle/DependencyInjection/Compiler/DataCollectorCompilerPass.php

```diff
@@ -24,7 +24,7 @@ class DataCollectorCompilerPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        if ($container->hasDefinition('profiler')) {
+        if ($container->hasDefinition('profiler') && $container->getParameter('kernel.environment') === 'dev') {
```

Comment:
Shouldn't we rely on `kernel.debug` parameter instead? 
As there could be multiple environments that use symfony profiler, it's not hardcoded to `dev` by Symfony itself.